.github/workflows/ccpp.yml

```diff
@@ -107,7 +108,7 @@ jobs:
       - name: Run C++ unit tests (MPI)
         run: |
           cd build/test
-          mpiexec -np 2 ctest -V --output-on-failure -R unittests
+          mpirun -n 3 --map-by :OVERSUBSCRIBE ctest -V --output-on-failure -R unittests
```

Comment:
Don't use MPI implementation specific args - we can't then switch MPI version. Use environment variables (we already use some OpenMPI env variables on the CI).

See https://github.com/open-mpi/ompi/issues/8955#issuecomment-846272943, https://github.com/open-mpi/ompi/issues/8955#issuecomment-1295456308 and https://github.com/mpi4py/mpi4py-testing/blob/1dcad02e04f8c65c3e31416dcb86a7b8c1affcb8/.github/workflows/openmpi.yml#L98.